### PR TITLE
Update package-lock.json to match package.json versions

### LINKED ISSUE #37316: Default starter package-lock.json has outdated dependency versions
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Following the steps of [Part 1 of the Gatsby Tutorial](https://www.gatsbyjs.com/docs/tutorial/part-1/) with Gatsby 5.3.3 creates a gatsby site that fails to deploy on Gatsby Cloud. This is because the generated package-lock.json file is out of date.

### Reproduction Link

https://github.com/gatsbyjs/gatsby

### Steps to Reproduce

Using gatsby v5.3.3 and with node and npm set up in the WSL per the instructions for Linux in [part 0 of the tutorial](https://www.gatsbyjs.com/docs/tutorial/part-0/):

1. gatsby new - accept all default options and install no additional features
2. Push this git repository to a Github repo, and from there deploy it to Gatsby Cloud using the instructions in [part 1 of the tutorial](https://www.gatsbyjs.com/docs/tutorial/part-1/)

### Expected Result

Site builds and deploys to Gatsby Cloud per the tutorial

### Actual Result

Deployments in Gatsby Cloud fails during the install dependencies step with Gatsby Cloud raw log error messages:

16:27:12 PM: npm ERR! code EUSAGE
16:27:12 PM: npm ERR! 
16:27:12 PM: npm ERR! `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.

This error can be fixed locally in the Gatsby site by deleting the package-lock.json file, running npm install and then pushing the updated code to Github.

### Environment

```shell
Gatsby environment info:
System:
OS: Linux 4.4 Ubuntu 20.04.5 LTS (Focal Fossa)
CPU: (8) x64 Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz
Shell: 5.0.17 - /bin/bash
Binaries:
Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
npm: 8.19.2 - ~/.nvm/versions/node/v18.12.1/bin/npm
npmPackages:
gatsby: ^5.3.3 => 5.3.3
npmGlobalPackages:
gatsby-cli: 5.3.1
```

### Config Flags

_No response_

### PR DESCRIPTION
**Pull request by raffishquartan**
## Description

Update package-lock.json so that dependencies match those in package.json - resulting code now builds and deploys to Gatsby Cloud successfully.

## Related Issues

Fixes #37316

### CODE REVIEW

File: starters/default/package-lock.json [hunk 1]
```diff
@@ -10,16 +10,14 @@
...
         "react-dom": "^18.2.0"
       },
       "devDependencies": {
-        "prettier": "^2.8.0"
+        "@types/node": "^18.11.10",
+        "@types/react": "^18.0.26",
+        "@types/react-dom": "^18.0.9",
+        "typescript": "^4.9.3"
       }
     },
```
Comment (pieh) [inline starters/default/package-lock.json:22]: This doesn't look right list of deps definitely doesn't match deps in package.json for this starter - https://github.com/gatsbyjs/gatsby/blob/250c8b799a88bc07c5cef18b66dc8ebe2cd1ad9e/starters/default/package.json#L7-L19

**Comment by pieh** [comment]
I opened https://github.com/gatsbyjs/gatsby/pull/37331 to see if lock files are out of date right now, if the lock files are good I think the issue might be somewhere else

File: starters/default/package-lock.json (hunk 1 above)
Comment (pieh) [reply starters/default/package-lock.json:22]: Ok, you are generally right about lock file being out of sync, just this PR changes to wrong one :)

I will be updating lock files in https://github.com/gatsbyjs/gatsby/pull/37331

**Comment by tyhopp** [comment]
Closing this since the issue is on Cloud side install (https://github.com/gatsbyjs/gatsby/issues/37316#issuecomment-1364166732)

**Pull request closed without merging**